src/github/provider.py

```diff
@@ -78,33 +79,33 @@ def get_patch(self, pr_file: dict) -> str:
         """
         return pr_file.get('patch', 'no changes')
 
-    def _to_PullRequest(self, owner: str, repo_name: str, pr: dict, username: str) -> PullRequest:
+    def _to_PullRequest(self, pr: dict) -> PullRequest:
         """
         Convert a GitHub pull request dictionary into a `PullRequest` dataclass.
 
         Args:
-            owner (str): The owner of the repository.
-            repo_name (str): The name of the repository.
             pr (dict): The pull request data from the GitHub API.
-            username (str): The username of the author to match.
 
         Returns:
             PullRequest: The `PullRequest` object containing detailed information about the PR.
         """
         pr_number = pr['number']
-        pr_files = self.api.pulls.list_files(owner=owner, repo=repo_name, pull_number=pr_number)
+        owner = pr['user']['login']
```

Comment:
is `owner` used in the following code? or do we use the one in the self?

Reply:
Nice catch